diff --git a/feature_creation.py b/feature_creation.py
index b549caa..6f01cd0 100644
--- a/feature_creation.py
+++ b/feature_creation.py
@@ -65,7 +65,7 @@ def feature_creation():
     logger.info(
         "Loading pre-mapped lat and long coordinates for customer's city and state, and using it to create 'cust_lat' and 'cust_long' columns."
     )
-    cust_loc = pd.read_csv(r"data/customer_locations.csv")
+    cust_loc = pd.read_parquet(r"data/cust_loc.parquet")
 
     df = df.merge(
         cust_loc[["city", "state", "cust_lat", "cust_long"]],
