diff --git a/feature_creation.py b/feature_creation.py
index 0d6baa8..6480421 100644
--- a/feature_creation.py
+++ b/feature_creation.py
@@ -112,8 +112,8 @@ def feature_creation():
     df["month"] = df["trans_dt"].dt.month
     df["day"] = df["trans_dt"].dt.day
     df["quarter"] = df["trans_dt"].dt.quarter
-    df["week"] = df["trans_dt"].dt.week
-    df["dayofweek"] = df["trans_dt"].dt.dayofweek
+    df["week"] = df["trans_dt"].dt.isocalendar().week
+    df["day_of_week"] = df["trans_dt"].dt.dayofweek
 
     # time related columns
     df["time"] = df["trans_dt"].dt.time
@@ -200,7 +200,7 @@ def feature_creation():
 
     ###creating 'age_at_trans' column
     logger.info("Creating age_at_trans column.")
-    df["age_at_trans"] = df["trans_dt"] - df["dob"].dt.days / 365.25
+    df["age_at_trans"] = (df["trans_dt"] - df["dob"]).dt.days / 365.25
 
     logger.info("Feature Creation Done")
 
