diff --git a/feature_creation.py b/feature_creation.py
index 0d6baa8..ec9f0ec 100644
--- a/feature_creation.py
+++ b/feature_creation.py
@@ -112,8 +112,8 @@ def feature_creation():
     df["month"] = df["trans_dt"].dt.month
     df["day"] = df["trans_dt"].dt.day
     df["quarter"] = df["trans_dt"].dt.quarter
-    df["week"] = df["trans_dt"].dt.week
-    df["dayofweek"] = df["trans_dt"].dt.dayofweek
+    df["week"] = df["trans_dt"].dt.isocalendar().week
+    df["day_of_week"] = df["trans_dt"].dt.dayofweek
 
     # time related columns
     df["time"] = df["trans_dt"].dt.time
